 modules/tv/classes/Channel.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/tv/classes/Channel.php b/modules/tv/classes/Channel.php
index ec8b368..ea9e6b7 100644
--- a/modules/tv/classes/Channel.php
+++ b/modules/tv/classes/Channel.php
@@ -121,7 +121,7 @@ class Channel extends MythBase {
         // Otherwise, grab it from the backend
             else {
             // Make the request and store the result
-                $data = MythBackend::find()->httpRequest('Guide/GetChannelIcon', array('ChanID' => $this->chanid));
+                $data = MythBackend::find()->httpRequest('Guide/GetChannelIcon', array('ChanId' => $this->chanid));
                 if ($data)
                     file_put_contents($this->icon, $data);
                 unset($data);
