diff --git a/includes/init.php b/includes/init.php
index 830223d..a0a9816 100644
--- a/includes/init.php
+++ b/includes/init.php
@@ -36,6 +36,9 @@
     require_once 'includes/errors.php';
     require_once 'includes/errordisplay.php';
 
+// Define some common stuff
+    require_once 'includes/defines.php';
+
 // Setup the database
     require_once 'includes/database.php';
 
@@ -45,9 +48,6 @@
 // Load the session handler routines
     require_once 'includes/session.php';
 
-// Define some common stuff
-    require_once 'includes/defines.php';
-
 // Load the translation routines so the modules can translate their descriptions
     require_once 'includes/translate.php';
